support/reattach-pv/main.go

```diff
@@ -207,13 +217,43 @@ type MatchingVolumeClaim struct {
 }
 
 // matchPVsWithClaim iterates over existing pvs to match them to an expected pvc.
-func matchPVsWithClaim(pvs []v1.PersistentVolume, claims map[types.NamespacedName]v1.PersistentVolumeClaim) ([]MatchingVolumeClaim, error) {
+func matchPVsWithClaim(pvs []v1.PersistentVolume, claims map[types.NamespacedName]v1.PersistentVolumeClaim, es esv1.Elasticsearch, oldESName string) ([]MatchingVolumeClaim, error) {
+	// The following regex defines the expected persistent volume claim name format.
+	// The regex patterns for both EsName, and NodeSetName groups are directly from apimachinery validation:
+	// https://github.com/kubernetes/apimachinery/blob/6809593a70331498e028baf9baa3b14b10c75214/pkg/util/validation/validation.go#L178
+	var (
+		r = regexp.MustCompile(fmt.Sprintf(`^elasticsearch-data-(%s)-es-([a-z0-9]([-a-z0-9]*[a-z0-9])?)-([0-9])+$`, oldESName))
+	)
+
 	matches := make([]MatchingVolumeClaim, 0, len(pvs))
 	for _, pv := range pvs {
 		if pv.Spec.ClaimRef == nil {
 			continue
 		}
-		claim, expected := claims[types.NamespacedName{Namespace: pv.Spec.ClaimRef.Namespace, Name: pv.Spec.ClaimRef.Name}]
+		expectedClaimName := pv.Spec.ClaimRef.Name
+		// if you're building a newly named cluster, from a previous cluster's PVs, we'll
+		// need to extract the nodeSetName, and ordinal from the PV's claimref.Name,
+		// and replace the expected claim name to be the newly generated cluster's name.
+		if oldESName != "" {
+			regexMatches := r.FindStringSubmatch(pv.Spec.ClaimRef.Name)
+			// 5 matches are expected here, as the first match is the full string, and the next is the ES name, NodeSetName+internal match, and Ordinal.
+			if len(regexMatches) != 5 {
+				continue
+			}
+			nodeSetName := regexMatches[2]
+			strOrdinal := regexMatches[4]
+			ordinal, err := strconv.Atoi(strOrdinal)
+			if err != nil {
```

Comment:
I would log this error to stderr since I don't think it is expected.

Reply:
I simply removed the print, as it's unnecessary.